proto/wtools/abase/l0/l1.test/Long.test.s

```diff
@@ -8754,6 +8754,238 @@ function longRight( test )
+
+  /* - */
+
+  test.case = 'no argument';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is();
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of no argument took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'null';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( null );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of null took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'function';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( function() {} );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of function took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'string';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( 'x' );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of string took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'number';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( 1 );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of number took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'boolean';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( false );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of boolean took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'object';
+  var took = 0;
+  var times = 50000000;
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( {} );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of object took : ${took / 1000}s` );
+
+  /* - */
+
+  test.case = 'object with fields and iteraor method';
+  var took = 0;
+  var times = 50000000;
+  var src = new function()
+  {
+    this[ Symbol.iterator ] = function ()
+    {
+      return { next() { return { done : true } } }
+    }
+  }
+  for( let i = times; i > 0; i-- )
+  {
+    var time = _.time.now();
+    var got = _.long.is( src );
+    took += _.time.now() - time;
+    //test.identical( got, false );
+  }
+  console.log( `${times} iterations of object with fields and iteraor method took : ${took / 1000}s` );
+}
```

Comment:
add subroutine to isolate code which is measured.

Reply:
Subroutine added:

function measure( test, got, expected, times, took )
  {
    test.identical( got, expected );
    console.log( `${times} iterations of ${test.case} took : ${took / 1000 }s on ${process.version}` );
  }